homeassistant/requirements.py

```diff
@@ -31,55 +37,77 @@ def __init__(self, domain: str, requirements: List) -> None:
         self.requirements = requirements
 
 
-async def async_get_integration_with_requirements(
-    hass: HomeAssistant, domain: str, done: Optional[Set[str]] = None
+async def _async_get_integration_with_requirements(
+    hass: "HomeAssistant",
+    domain: str,
+    cache: Dict[str, Any],
+    event: asyncio.Event,
+    args: Optional[Dict[str, Any]],
 ) -> Integration:
     """Get an integration with all requirements installed, including the dependencies.
 
-    This can raise IntegrationNotFound if manifest or integration
-    is invalid, RequirementNotFound if there was some type of
-    failure to install requirements.
+    Update the passed in cache and resolve the load event.
     """
+    done: Optional[Set] = cast(Set, args.get("done")) if args else None
     if done is None:
         done = {domain}
     else:
         done.add(domain)
 
     integration = await async_get_integration(hass, domain)
 
-    if hass.config.skip_pip:
-        return integration
-
-    if integration.requirements:
-        await async_process_requirements(
-            hass, integration.domain, integration.requirements
-        )
-
-    deps_to_check = [
-        dep
-        for dep in integration.dependencies + integration.after_dependencies
-        if dep not in done
-    ]
-
-    for check_domain, to_check in DISCOVERY_INTEGRATIONS.items():
-        if (
-            check_domain not in done
-            and check_domain not in deps_to_check
-            and any(check in integration.manifest for check in to_check)
-        ):
-            deps_to_check.append(check_domain)
-
-    if deps_to_check:
-        await asyncio.gather(
-            *[
-                async_get_integration_with_requirements(hass, dep, done)
-                for dep in deps_to_check
-            ]
-        )
+    if not hass.config.skip_pip:
```

Comment:
Please keep this as a guard clause.

```python
if not hass.config.skip_pip:
    cache[domain] = integration
    event.set()
    return
```

Reply:
Ok. I actually had it that way at first, but refactored to remote the need to duplicate the `cache[domain] = integration; event.set()` statements.